### PR TITLE
New Adapter: Adview

### PR DESCRIPTION
**Pull request by And1sS**

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final Imp firstImp = request.getImp().get(0);
+        final ExtImpAdview extImpAdview;
+
+        try {
+            extImpAdview = mapper.mapper().convertValue(firstImp.getExt(), ADVIEW_EXT_TYPE_REFERENCE).getBidder();
+        } catch (DecodeException e) {
+            return Result.withError(BidderError.badInput(e.getMessage()));
```
[the remark this answers is not in the document]
Comment (snahornyi) [reply src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java:56]: Pls cover it with UT

**Review by snahornyi — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final Imp firstImp = request.getImp().get(0);
+        final ExtImpAdview extImpAdview;
+
+        try {
+            extImpAdview = mapper.mapper().convertValue(firstImp.getExt(), ADVIEW_EXT_TYPE_REFERENCE).getBidder();
+        } catch (DecodeException e) {
```
[the remark this answers is not in the document]
Comment (And1sS) [reply src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java:55]: I'd say we don't need a separate method, cuz makeHttpRequests is simple enough to contain this type of logic.

File: src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                .method(HttpMethod.POST)
+                .uri(resolveEndpoint(accountId))
+                .headers(HttpUtil.headers())
+                .body(mapper.encode(request))
+                .payload(request)
+                .build();
+    }
+
+    private String resolveEndpoint(String accountId) {
+        return endpointUrl.replace(ACCOUNT_ID_MACRO, HttpUtil.encodeUrl(accountId));
```
[the remark this answers is not in the document]
Comment (And1sS) [reply src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java:104]: I disagree, here we are modifying url with unknown string, it can be anything and can contain any character.. i'd say it's a barrier to make sure that resulting url is ok.

File: src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (And1sS) [reply src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java:56]: Ok, but should i cover it with UT? this should never happen since pbs core validates extension before it gets to this point.. maybe just get rid of this try catch?

File: src/test/java/org/prebid/server/bidder/adview/AdviewBidderTest.java [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+
+        // when
+        final Result<List<BidderBid>> result = adviewBidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue())
+                .containsExactly(BidderBid.of(Bid.builder().impid("123").build(), banner, null));
+    }
+
```
Comment (snahornyi) [inline src/test/java/org/prebid/server/bidder/adview/AdviewBidderTest.java:221]: As discussed before pls add 
```java
    @Test
    public void makeHttpRequestsShouldReturnErrorsOfNotValidImps() {
        // given
        final BidRequest bidRequest = givenBidRequest(
                impBuilder -> impBuilder
                        .ext(mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode()))));
        // when
        final Result<List<HttpRequest<BidRequest>>> result = adviewBidder.makeHttpRequests(bidRequest);

        // then
        assertThat(result.getErrors()).hasSize(1);
        assertThat(result.getErrors())
                .containsExactly(BidderError.badInput("invalid imp.ext"));
    }
```
Also pls add test for native bid and also for case for first format is null

**Review by snahornyi — CHANGES_REQUESTED** [review]

**Review by snahornyi — APPROVED** [review]
